engine/src/sysspec-url.cpp

```diff
@@ -215,91 +220,172 @@ struct MCSGetUrlState
-        }
-        else
-        /* UNCHECKED */ MCDataCreateMutable(0, &t_new_data);
-        
-        /* UNCHECKED */ MCDataAppend(*t_new_data, (MCDataRef)p_data);
-        /* UNCHECKED */ MCDataCopy(*t_new_data, context -> data);
-        send_url_progress(context -> object, p_status, context -> url, MCDataGetLength(context -> data), context -> total, p_data);
-    }
-	return true;
+	MCSGetUrlState *context = static_cast<MCSGetUrlState *>(p_context);
+
+	/* Update request status */
+	context->status = p_status;
+
+	/* Handle the different possible interpretations of the data
+	 * argument. */
+	if (p_data != nil)
+	{
+		switch(p_status)
+		{
+		case kMCSystemUrlStatusError:
+			/* The data is an error message */
+			MCValueAssign(context->error, (MCStringRef) p_data);
+			break;
+
+		case kMCSystemUrlStatusNegotiated:
+			/* The data is the total download size */
+			context->total_size = *(int32_t *) p_data;
+			break;
+
+		case kMCSystemUrlStatusLoading:
+			/* The data is a chunk of the URL's content.  Append it to
+			 * the current download buffer */
+			if (context->data == nil)
+			{
+				if (!MCDataCopy((MCDataRef) p_data, context->data))
+				{
+					return false;
+				}
+			}
+			else if (!MCDataIsEmpty((MCDataRef) p_data))
+			{
+				if (!MCDataMutableCopyAndRelease(context->data, context->data))
+				{
+					return false;
+				}
+
+				if (!MCDataAppend(context->data, (MCDataRef) p_data))
+				{
+					return false;
+				}
+
+				if (!MCDataCopyAndRelease(context->data, context->data))
+				{
+					return false;
+				}
+			}
+
+			/* Update the total amount loaded */
+			context->loaded_size = MCMax(context->loaded_size,
+			                             MCDataGetLength(context->data));
+			break;
 
+		case kMCSystemUrlStatusLoadingProgress:
+			/* The data is the total amount loaded so far */
+			context->loaded_size = MCMax(context->loaded_size,
+			                             *(uint32_t *) p_data);
+			break;
+
+		default:
+			/* Ignore any attached data */
+			break;
+		}
+	}
+
+	/* Send an message with the latest loading progress */
+	MCUrlProgressEvent *t_event;
+	t_event = MCUrlProgressEvent::CreateUrlProgressEvent(
+	            context->object, context->url, context->status,
+	            context->loaded_size, context->total_size,
+	            context->error);
+	if (t_event == nil)
+	{
+		return false;
+	}
+
+	MCEventQueuePostCustom(t_event);
+	return true;
 }
 
-void MCS_geturl(MCObject *p_target, MCStringRef p_url)
+void
+MCS_geturl(MCObject *p_target,
+           MCStringRef p_url)
 {
+	// Preprocess the URL by removing leading and trailing whitespace
 	MCAutoStringRef t_processed_url;
-	
-	// IM-2013-07-30: [[ Bug 10800 ]] strip whitespace chars from url before attempting to fetch
-	if (!MCSystemProcessUrl(p_url, kMCSystemUrlOperationStrip, &t_processed_url))
+	if (!MCSystemProcessUrl(p_url, kMCSystemUrlOperationStrip,
+	                        &t_processed_url))
+	{
 		return;
-	
```

Comment:
What is the purpose of changing the formatting and removing the bugfix comment here?